pkg/broker/filter/server_manager.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filter
+
+import (
+	"context"
+	"crypto/tls"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/eventing/pkg/eventingtls"
+	"knative.dev/eventing/pkg/kncloudevents"
+	"knative.dev/pkg/configmap"
+
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	secretinformer "knative.dev/pkg/injection/clients/namespacedkube/informers/core/v1/secret"
+)
+
+func NewServerManager(ctx context.Context, logger *zap.Logger, cmw configmap.Watcher, httpPort, httpsPort int, handler *Handler) (*eventingtls.ServerManager, error) {
+	tlsConfig, err := getServerTLSConfig(ctx)
+	if err != nil {
+		logger.Info("failed to get TLS server config", zap.Error(err))
+	}
+
+	httpReceiver := kncloudevents.NewHTTPMessageReceiver(httpPort)
+	httpsReceiver := kncloudevents.NewHTTPMessageReceiver(httpsPort, kncloudevents.WithTLSConfig(tlsConfig))
+
+	return eventingtls.NewServerManager(ctx, httpReceiver, httpsReceiver, handler, cmw)
+}
+
+func getServerTLSConfig(ctx context.Context) (*tls.Config, error) {
+	secret := types.NamespacedName{
+		Namespace: "knative-eventing",
+		Name:      "mt-broker-filter-server-tls", // wasn't it: "mt-channel-based-broker-filter-tls"???
```

Comment:
I am not sure that we are reconciling `mt-broker-filter-server-tls` secret in our broker reconciler 
\cc @pierDipi 

https://github.com/knative/eventing/blob/2a5a9a5134d4b08c5e03c8c1d26220e4aa44a7af/pkg/reconciler/broker/broker.go#L60

Reply:
/cc @pierDipi

Reply:
I'm working on it

Reply:
regarding
> wasn't it: "mt-channel-based-broker-filter-tls"???

We're using
https://github.com/knative/eventing/blob/2a5a9a5134d4b08c5e03c8c1d26220e4aa44a7af/config/brokers/mt-channel-broker-tls/broker-filter-tls-certificate.yaml#L22

Reply:
yes, I was just "confused" initially, because in the TLS proposal it was named differently: https://docs.google.com/document/d/1H-x_oji8LqkCyd7tlsSyclmUe7FAmEJPgRxOU_0pkn8/edit?resourcekey=0-lzDIPJsZOP3G17QE_g1lHw#heading=h.2rykv8r1mpzb ...

Reply:
Yes, it's more aligned to deployments names now